octodns/provider/easydns.py

```diff
@@ -0,0 +1,453 @@
+
+    def __init__(self):
+        super(EasyDNSClientUnauthorized, self).__init__('Unauthorized')
+
+
+class EasyDNSClient(object):
+    # EasyDNS Sandbox API
+    SANDBOX = 'https://sandbox.rest.easydns.net'
+    # EasyDNS Live API
+    LIVE = 'https://rest.easydns.net'
+    # Default Currency CAD
+    defaultCurrency = 'CAD'
+    # Domain Portfolio
+    domainPortfolio = 'myport'
+
+    def __init__(self, token, apikey, currency, portfolio, sandbox):
+        self.log = logging.getLogger('EasyDNSProvider[{}]'.format(id))
+        self.token = token
+        self.apikey = apikey
+        self.defaultCurrency = currency
+        self.domainPortfolio = portfolio
+        self.apienv = 'sandbox' if sandbox else 'live'
+        authkey = '{}:{}'.format(self.token, self.apikey)
+        self.authkey = base64.b64encode(authkey.encode("utf-8"))
+        self.basepath = self.SANDBOX if sandbox else self.LIVE
+        sess = Session()
+        sess.headers.update({'Authorization': 'Basic {}'.format(self.authkey)})
+        sess.headers.update({'accept': 'application/json'})
+        self._sess = sess
+
+    def _request(self, method, path, params=None, data=None):
+        url = '{}{}'.format(self.basepath, path)
+        resp = self._sess.request(method, url, params=params, json=data)
+        if resp.status_code == 400:
+            self.log.debug('Response code 400, path=%s', path)
+            if method == 'GET' and path[:8] == '/domain/':
+                raise EasyDNSClientNotFound()
+            raise EasyDNSClientBadRequest()
+        if resp.status_code == 401:
+            raise EasyDNSClientUnauthorized()
+        if resp.status_code == 403 or resp.status_code == 404:
+            raise EasyDNSClientNotFound()
+        resp.raise_for_status()
+        return resp
+
+    def domain(self, name):
+        path = '/domain/{}'.format(name)
+        return self._request('GET', path).json()
+
+    def domain_create(self, name):
+        # EasyDNS allows for new domains to be created for the purpose of DNS
+        # only, or with domain registration. This function creates a DNS only
+        # record expectig the domain to be registered already
+        path = '/domains/add/{}'.format(name)
+        domainData = {'service': 'dns',
+                      'term': 1,
+                      'dns_only': 1,
+                      'portfolio': self.domainPortfolio,
+                      'currency': self.defaultCurrency}
+        self._request('PUT', path, data=domainData).json()
+
+        # EasyDNS creates default records for MX, A and CNAME for new domains,
+        # we need to delete those default record so we can sync with the source
+        # records, first we'll sleep for a second before gathering new records
+        # We also create default NS records, but they won't be deleted
+        sleep(1)
+        records = self.records(name, True)
+        for record in records:
+            if record['type'] in ('A', 'MX', 'CNAME'):
```

Comment:
Would it make sense to check the `host` part of the records too here just to be double sure it's deleting the things it expects to be?

Reply:
Good idea just in case the API author decides to make some changes without informing me.  This has been added to the next commit.